st2common/st2common/util/mongoescape.py

```diff
@@ -71,14 +73,20 @@ def _translate_chars(field, translation):
 def escape_chars(field):
     # TODO: This is slow on large dicts, figure out if we can safely manipulate original value
     # instead of creating the copy
-    value = copy.deepcopy(field)
+    if not isinstance(field, dict):
+        return field
+
+    value = ujson.loads(ujson.dumps(field))
```

Comment:
It's quite a difference, especially when working with large documents.

```bash
Timer unit: 1e-06 s

Total time: 0.330095 s
File: /data/stanley/st2common/st2common/util/mongoescape.py
Function: escape_chars_1 at line 73

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
    73                                           @profile
    74                                           def escape_chars_1(field):
    75                                               # TODO: This is slow on large dicts, figure out if we can safely manipulate original value
    76                                               # instead of creating the copy
    77         4         11.0      2.8      0.0      if not isinstance(field, dict):
    78                                                   return field
    79                                           
    80         4     280657.0  70164.2     85.0      value = copy.deepcopy(field)
    81                                           
    82                                               #value = ujson.loads(ujson.dumps(field))
    83         4      49427.0  12356.8     15.0      return _translate_chars(value, ESCAPE_TRANSLATION)



Timer unit: 1e-06 s

Total time: 0.052098 s
File: /data/stanley/st2common/st2common/util/mongoescape.py
Function: escape_chars_1 at line 73

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
    73                                           @profile
    74                                           def escape_chars_1(field):
    75                                               # TODO: This is slow on large dicts, figure out if we can safely manipulate original value
    76                                               # instead of creating the copy
    77         4          6.0      1.5      0.0      if not isinstance(field, dict):
    78                                                   return field
    79                                           
    80         4       7437.0   1859.2     14.3      value = ujson.loads(ujson.dumps(field))
    81         4      44655.0  11163.8     85.7      return _translate_chars(value, ESCAPE_TRANSLATION)

Total time: 0.052882 s
File: test_escape.py
Function: escape1 at line 132
```


Problem is there are some edge cases, so I'm not sure yet if I can get it to fully work - 5af09e82891aa93d35d584ae988ed9050aba7874, 4917d47b1f7fc181a5d5944286a208330d4a7698, 67886184b511cb11b5f0c4f239be937c21124ef6.